pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1115,60 +874,6 @@ func TestDeleteControllerAndExpectations(t *testing.T) {
 	validateSyncReplicaSet(t, &fakePodControl, 0, 0, 0)
 }
 
-// shuffle returns a new shuffled list of container controllers.
-func shuffle(controllers []*extensions.ReplicaSet) []*extensions.ReplicaSet {
-	numControllers := len(controllers)
-	randIndexes := rand.Perm(numControllers)
-	shuffled := make([]*extensions.ReplicaSet, numControllers)
-	for i := 0; i < numControllers; i++ {
-		shuffled[i] = controllers[randIndexes[i]]
-	}
-	return shuffled
-}
-
-func TestOverlappingRSs(t *testing.T) {
```

Comment:
This should also stay as a unit test, but refactored around a function that just outputs which RSes to enqueue in response to a given addPod event. We can't observe if RSes get needlessly enqueued in an integration test.

Reply:
There is an integration test named `TestOverlappingRSs` for this, so i remove this test. You are definitely right that the integration test does not fully cover this unit test. There is a need to check which RSs are enqueued due to an event.